packages/template-retail-react-app/app/hooks/use-add-to-cart-modal.test.js

```diff
@@ -582,7 +589,11 @@ test('Renders AddToCartModal', () => {
         </AddToCartModalContext.Provider>
     )
 
-    expect(getByText(MOCK_PRODUCT.name)).toBeInTheDocument()
+    expect(screen.getAllByText(MOCK_PRODUCT.name)[0]).toBeInTheDocument()
+    expect(screen.getByRole('dialog', {name: /23 items added to cart/i})).toBeInTheDocument()
+
+    const numOfRowsRendered = screen.getAllByTestId('product-added').length
+    expect(numOfRowsRendered).toEqual(MOCK_DATA.itemsAdded.length)
```

Comment:
I remember you saying something about "awaiting too many things" which I'm still a bit confused about. If you want all of the `expect` conditions to be true at the same time (usually true, I believe) then I would wrap this in `waitFor(() => {})`

Reply:
I think I won't use the `waitFor` in this case (in the product-sets feature branch). Maybe it becomes necessary within the hooks-integration work. 

After the component is rendered, I expect all those `expect` conditions to be true.. all at the same time, yup. Because one render of the component is enough. It won't be re-rendered by some data fetching or async operation.